deploy_docs.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+set -e # Exit with nonzero exit code if anything fails
+
+# Build the documentation from the SOURCE_BRANCH
+# and push it to TARGET_BRANCH.
+SOURCE_BRANCH="master"
+TARGET_BRANCH="gh-pages"
+
+# Save some useful information
+REPO=`git config remote.origin.url`
+SSH_REPO=${REPO/https:\/\/github.com\//git@github.com:}
+SHA=`git rev-parse --verify HEAD`
+
+# Clone the existing gh-pages for this repo into out/
+# Create a new empty branch if gh-pages doesn't exist yet (should only happen on first deply)
+git clone $REPO out
+cd out
+
+git checkout $TARGET_BRANCH || git checkout --orphan $TARGET_BRANCH
+cd ..
+
+# Clean out existing contents
+rm -rf out/sphinx_docs/ || exit 0
+
+# Pull from SOURCE_BRANCH again
+git pull origin $SOURCE_BRANCH
+
+# Build the Sphinx documentation
+cd docs
+make html
+cd ../  
+
+mkdir -p out/sphinx_docs/
+mv -f docs/_build/html/* out/sphinx_docs/
+touch out/.nojekyll
+
+# Now let's go have some fun with the cloned repo
+cd out
+git config --local user.name "Azure Pipelines"
+git config --local user.email "azuredevops@microsoft.com"
+
+echo "doing git add/commit/push"
+
+# Commit the "changes", i.e. the new version.
+# The delta will show diffs between new and old versions.
+git add --all
+
+# Exit if there are no docs changes
+if git diff --staged --quiet; then
+   echo "exiting with no docs changes"
+   exit 0
+fi
+
+# Otherwise, commit and push
+git commit -m "Deploy to GitHub Pages: ${SHA}"
+git push $SSH_REPO $TARGET_BRANCH
+cd ..
+
+# See https://github.com/AMReX-Codes/amrex/blob/master/LICENSE for the license
```

Comment:
what LICENSE is this. just add it on top (MIT, BSD3, etc. ) - and say this file was modified from ... and is LICENSed the same

Reply:
The license is the LBNL variant of the BSD3 license: https://spdx.org/licenses/BSD-3-Clause-LBNL.html

Reply:
The original script is this one https://github.com/AMReX-Codes/amrex/blob/master/build_and_deploy.sh

Reply:
I would suggest that we say at the beginning of the file that we take this from  https://github.com/AMReX-Codes/amrex/blob/master/build_and_deploy.sh licensed accoriding to  https://spdx.org/licenses/BSD-3-Clause-LBNL.html - maybe

Reply:
That's probably fine, but better might be to link to the repo's DOI (https://doi.org/10.5281/zenodo.2555438), and say that it's licensed according to BSD-3-Clause-LBNL (https://github.com/AMReX-Codes/amrex/blob/master/LICENSE).